plugins/item_tasks/web_client/templates/paginateTasksWidget.pug

```diff
@@ -3,6 +3,6 @@
 
 .list-group.g-task-list-container
   each task in tasks
-    a.list-group-item.g-execute-task-link(href=`#item_task/${task.id}/run`)
+    .list-group-item.g-execute-task-link(data-task-id=task.id)
```

Comment:
Just to chime in, I think that having widgets like this only fire `g:selected` events, then letting the parent class hook up routing, provides a very clean abstraction mechanism and allows for reuse of these widgets with minimal dependencies.

Of course, at the end of the day, user experience should take precedence over developer convenience. So, perhaps the `PaginateTasksWidget` could take an optional parameter, which would be a callback function, which if provided, should take a single `task` argument (of an item_task model) and return a string (of a URL to be used as the task link `href`). The `PaginateTasksWidget` should still always continue to emit `g:selected` as before.